pkg/pidhandle/pidhandle_linux_test.go

```diff
@@ -0,0 +1,235 @@
+	defer func() { pidfdOpen = original }()
+	pidfdOpen = func(pid int, flags int) (int, error) {
+		return 255, nil
+	}
+
+	// Mock the nameToHandleAt
+	original_nameToHandleAt := nameToHandleAt
+	defer func() { nameToHandleAt = original_nameToHandleAt }()
+	nameToHandleAt = func(dirfd int, path string, flags int) (handle unix.FileHandle, mountID int, err error) {
+		return newFileHandle(254, []byte("test")), 1, nil
+	}
+
+	h, err := NewPIDHandle(os.Getpid())
+	assert.NoError(t, err)
+	defer h.Close()
+
+	pidData, err := h.String()
+	assert.NoError(t, err)
+	assert.Equal(t, nameToHandlePrefix+"254 74657374", pidData)
+	assert.Equal(t, h.PID(), os.Getpid())
+}
+
+func TestNewPIDHandlepidfdOpenNotSupported(t *testing.T) {
+	// Mock the pidfdOpen
+	original := pidfdOpen
+	defer func() { pidfdOpen = original }()
+	pidfdOpen = func(pid int, flags int) (int, error) {
+		return -1, unix.ENOSYS
+	}
+
+	h, err := NewPIDHandle(os.Getpid())
+	assert.NoError(t, err)
+	defer h.Close()
+
+	pidData, err := h.String()
+	assert.NoError(t, err)
+	assert.Contains(t, pidData, startTimePrefix)
+	assert.Equal(t, h.PID(), os.Getpid())
+}
+
+func TestPIDHandleStringnameToHandleAtNotSupported(t *testing.T) {
+	// Mock the pidfdOpen
+	original := pidfdOpen
+	defer func() { pidfdOpen = original }()
+	pidfdOpen = func(pid int, flags int) (int, error) {
+		return 254, nil
+	}
+
+	// Mock the nameToHandleAt
+	original_nameToHandleAt := nameToHandleAt
+	defer func() { nameToHandleAt = original_nameToHandleAt }()
+	nameToHandleAt = func(dirfd int, path string, flags int) (handle unix.FileHandle, mountID int, err error) {
+		return newFileHandle(-1, []byte("")), 1, unix.ENOTSUP
+	}
+
+	h, err := NewPIDHandle(os.Getpid())
+	assert.NoError(t, err)
+	defer h.Close()
+
+	pidData, err := h.String()
+	assert.NoError(t, err)
+	assert.Contains(t, pidData, startTimePrefix)
+}
+
+func TestNewPIDHandleFromString(t *testing.T) {
+	// Mock the pidfdOpen
+	original := pidfdOpen
+	defer func() { pidfdOpen = original }()
+	pidfdOpen = func(pid int, flags int) (int, error) {
+		return 254, nil
+	}
+
+	// Mock the newFileHandle
+	original_newFileHandle := newFileHandle
+	defer func() { newFileHandle = original_newFileHandle }()
+	newFileHandle = func(fhType int32, bytes []byte) unix.FileHandle {
+		return unix.NewFileHandle(254, []byte("test"))
+	}
+
+	// Mock the openByHandleAt
+	original_openByHandleAt := openByHandleAt
+	defer func() { openByHandleAt = original_openByHandleAt }()
+	openByHandleAt = func(mountFD int, handle unix.FileHandle, flags int) (fd int, err error) {
+		return 255, nil
+	}
+
+	// Mock the nameToHandleAt
+	original_nameToHandleAt := nameToHandleAt
+	defer func() { nameToHandleAt = original_nameToHandleAt }()
+	nameToHandleAt = func(dirfd int, path string, flags int) (handle unix.FileHandle, mountID int, err error) {
+		return newFileHandle(255, []byte("test")), 1, nil
+	}
+
+	h, err := NewPIDHandleFromString(os.Getpid(), nameToHandlePrefix+"254 74657374")
```

Comment:
if you are going to push a new version, could you add a test for the empty string?